### DISCUSSION TITLE
Dyanamic Variables for REST Queries Not Passed?

### DISCUSSION THREAD

**Discussion by guekling** [Budibase/budibase#11261] [Help] [answered]
I have two REST endpoints:
(1) Get Access Token, where I create a dynamic variable based on my response body.
![image](https://github.com/Budibase/budibase/assets/24616618/0e80aeaf-909e-432c-b5bc-8729056a1080)
![image](https://github.com/Budibase/budibase/assets/24616618/1b34cbb3-e5a3-4a1e-bee7-5e2f4eead572)
(2) Get Objects, which requires a bearer authorization. The access token is retrieved from the dynamic variable.
![image](https://github.com/Budibase/budibase/assets/24616618/d5c9e29d-d0c4-4dac-8e05-bcbce3c7be43)

This had worked fine the last week, but after putting my app aside for a few days, I realised that it no longer works now.
When I send the "Get Objects" request, this is thrown because `POST {base_url}/api/queries/preview` is throwing a 400 error:
![image](https://github.com/Budibase/budibase/assets/24616618/9c84f08d-2b7b-49e1-bb99-3a76cc17927e)
However, it does not hit the "Get Objects" endpoint at all.

Simply changing the `Bearer {{ Dynamic.access_token }}` to `Bearer {{ Dynamic.Dynamic.access_token }}` allows it to hit the endpoint, but what I am receiving is `Bearer ` on the endpoint.

Would anyone have any idea what I'm doing wrong? Thank you!

**Comment by andz-bb** [comment, marked as answer]
hey @guekling, thanks for the detailed post! I have been able to recreate this on my end and have logged it as a bug here: #11285 

I'll discuss it with our engineers later today, hopefully we can get a fix out for you soon :+1:

**Comment by julia-szoradi** [comment]
Hi,

We upgraded to the latest version (2.9.36) and recreated the dynamic variables to fix any duplication that occurred in 2.8.10.
But we still face the same issue described above. The dynamic variable created from the results of the first query is not passed on to the second query.

From the request logs on our server, we can see that we receive `Bearer ` on the endpoint, so `{{ Dynamic.access_token }}` is evaluated to an empty string. While a few months ago the same app worked without issues.

**Comment by buanet** [comment]
Hi,
I'm facing the same issue with 2.10.6...

First Query (POST) - Get session token
![image](https://github.com/Budibase/budibase/assets/16137672/0e903281-7066-46db-8561-8b2270eac26a)

![image](https://github.com/Budibase/budibase/assets/16137672/7edb0c47-aa4d-4571-bcf3-196f334af635)

Second Query (GET) - Request data with session token in Headers

![image](https://github.com/Budibase/budibase/assets/16137672/c841a2ca-69bc-4ac0-b603-db850a016782)

Response:

`Authorization header is either missing or with an empty session token value.` (401)

Putting in the sessionToken directly into the second query works. So i think (hope) I configured it corretly. ;)
Looks like the dynamic variable is not passed between both queries.

Regards,
André
